File: sdk/flutter-plugin/lib/src/cesium_camera_state.dart
class CesiumCameraState {
  const CesiumCameraState({
    required this.longitude,
    required this.latitude,
    required this.altitudeMeters,
    required this.autoOrbit,
    this.bearing = 0,
    this.pitch = 0,
  });

  final double longitude;
  final double latitude;
  final double altitudeMeters;
  final bool autoOrbit;
  final double bearing;
  final double pitch;

  Map<String, Object?> toMap() => {
    'longitude': longitude,
    'latitude': latitude,
    'altitudeMeters': altitudeMeters,
    'autoOrbit': autoOrbit,
    'bearing': bearing,
    'pitch': pitch,
  };

  factory CesiumCameraState.fromMap(Map<String, Object?> map) {
    double asDouble(String key) => (map[key] as num?)?.toDouble() ?? 0;

    return CesiumCameraState(
      longitude: asDouble('longitude'),
      latitude: asDouble('latitude'),
      altitudeMeters: asDouble('altitudeMeters'),
      autoOrbit: map['autoOrbit'] as bool? ?? false,
      bearing: asDouble('bearing'),
      pitch: asDouble('pitch'),
    );
  }
}

File: sdk/android-sdk/src/main/kotlin/com/example/cesiumpoc/cesium_native_android_poc/CesiumCameraState.kt
package com.example.cesiumpoc.cesium_native_android_poc

/**
 * WGS84 camera contract shared by Android hosts and the Flutter bridge.
 *
 * `longitude` and `latitude` are the WGS84 view target. `altitudeMeters` is
 * the camera range from that target in meters. Tile selection and raster
 * overlay management remain inside cesium-native; the host only passes camera
 * state into the renderer backend.
 */
data class CesiumCameraState(
    val longitude: Double,
    val latitude: Double,
    val altitudeMeters: Double,
    val autoOrbit: Boolean,
    val bearing: Double = 0.0,
    val pitch: Double = 0.0,
)
